examples/tutorial/x/blog/handlers_test.go

```diff
@@ -0,0 +1,466 @@
+package blog
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/x"
+	"github.com/stretchr/testify/require"
+)
+
+var newTx = x.TestHelpers{}.MockTx
+
+func newContextWithAuth(addr string) (weave.Context, x.Authenticator) {
+	helpers := x.TestHelpers{}
+	ctx := context.Background()
+	// Set current block height to 100
+	ctx = weave.WithHeight(ctx, 100)
+	auth := helpers.CtxAuth("authKey")
+	// Create a new context and add addr to the list of signers
+	return auth.SetConditions(ctx, weave.Condition(weave.NewAddress([]byte(addr)))), auth
+}
+
+func newTestHandler(name string, auth x.Authenticator) weave.Handler {
+	switch name {
+	case "CreateBlogMsgHandler":
+		return CreateBlogMsgHandler{
+			auth:   auth,
+			bucket: NewBlogBucket(),
+		}
+	case "CreatePostMsgHandler":
+		return CreatePostMsgHandler{
+			auth:  auth,
+			blogs: NewBlogBucket(),
+			posts: NewPostBucket(),
+		}
+	case "RenameBlogMsgHandler":
+		return RenameBlogMsgHandler{
+			auth:   auth,
+			bucket: NewBlogBucket(),
+		}
+	case "ChangeBlogAuthorsMsgHandler":
+		return ChangeBlogAuthorsMsgHandler{
+			auth:   auth,
+			bucket: NewBlogBucket(),
+		}
+	case "SetProfileMsgHandler":
+		return SetProfileMsgHandler{
+			auth:   auth,
+			bucket: NewProfileBucket(),
+		}
+	default:
+		panic(fmt.Errorf("newTestHandler: unknown handler"))
+	}
+}
+func TestCreateBlogMsgHandlerCheck(t *testing.T) {
+	ctx, auth := newContextWithAuth("3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A")
+	testcases := []struct {
+		handler CreateBlogMsgHandler
+		msg     CreateBlogMsg
+		res     weave.CheckResult
+	}{
+		{
+			handler: newTestHandler("CreateBlogMsgHandler", auth).(CreateBlogMsgHandler),
+			msg: CreateBlogMsg{
+				Slug:    "this_is_a_blog",
+				Title:   "this is a blog title",
+				Authors: [][]byte{x.MainSigner(ctx, auth).Address()},
+			},
+			res: weave.CheckResult{
+				GasAllocated: newBlogCost,
+			},
+		},
+	}
+
+	for _, test := range testcases {
+		db := store.MemStore()
+		res, err := test.handler.Check(ctx, db, newTx(&test.msg))
+		require.NoError(t, err)
+		require.Equal(t, newBlogCost, res.GasAllocated,
+			fmt.Sprintf("gas allocated cost was equal to %d", res.GasAllocated))
+	}
+}
+
+func TestCreateBlogMsgHandlerDeliver(t *testing.T) {
+	ctx, auth := newContextWithAuth("3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A")
+	testcases := []struct {
+		handler CreateBlogMsgHandler
+		msg     CreateBlogMsg
+		obj     Blog
+	}{
+		{
+			handler: newTestHandler("CreateBlogMsgHandler", auth).(CreateBlogMsgHandler),
+			msg: CreateBlogMsg{
+				Slug:    "this_is_a_blog",
+				Title:   "this is a blog title",
+				Authors: [][]byte{x.MainSigner(ctx, auth).Address()},
```

Comment:
This seems a bit confusing to me. This is basically the string 
"3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A", right? Can't you just create `signer` on the top, and pass that both into `auth` as well as `Authors`